jsonschema-hyperschema.xml

```diff
@@ -124,152 +129,235 @@
+                    </t>
+                    <t>
+                        These links MUST all be presented as a single set, regardless
+                        of the number of subschemas involved.  The order of links in the set
+                        is not significant.
+                    </t>
+                </section>
+                <section title="By link context location">
+                    <t>
+                        Implementations MUST also make accessible the set of links for which
+                        a given instance location is the link's context.  All such links MUST
+                        be in at least one set of links produced by examining schema applicability.
+                    </t>
+                    <t>
+                        Link sets based on a context can include links attached to elements
+                        of an ordered array.  Implementations MUST preserve the order of
+                        such links with respect to each other.  Order is not otherwise significant.
+                    </t>
+                </section>
+            </section>
+            <section title="Link usage">
                 <t>
-                    Hyper-schema keywords in a subschema contained within a "not", at any depth,
-                    including any number of intervening additional "not" subschemas, MUST be
-                    ignored.
+                    Once discovered, a link is generally selected based on its link relation
+                    type, and used by communicating with the resource at the target URI.
+                    This means that the target URI must be correctly computed based on the
+                    instance data and optionally on input.  The context URI must similarly
+                    be calculated, although often only for use in context-based discovery
+                    as described above.  Finally, implementations need to recognize link
+                    relation types that have well-known semantics within JSON Hyper-Schema.
                 </t>
+                <section title="Target and context identification">
+                    <t>
+                        For a given link, an implementation MUST be able to identify the
+                        link's context.  The identifier MUST either be an IRI or URI
+                        that fully identifies the context, or an IRI or URI together
+                        with a JSON Pointer if the context is not an entire resource
+                        and the representation's media type does not allow constructing an
+                        appropriate fragment.
+                    </t>
+                    <t>
+                        For a given link, an implementation MUST be able to construct the
+                        target IRI or URI based on the instance and, if "hrefSchema" is present,
+                        client input.  When working with client input, the input data MUST be
```

Comment:
@dlax yeah, makes sense.  Trying to think of how to describe it other than "and, if allowed, client input".  I'm not even sure "allowed" is right.  "requested" might be more accurate, as "hrefSchema" is a pretty strong hint that the client should at least provide the option of input, whatever that means in the context of the client application.

I want to avoid the "form" terminology, though, as it has tended to cause confusion.